staging/src/k8s.io/component-base/metrics/registry_test.go

```diff
@@ -204,12 +207,6 @@ func TestMustRegister(t *testing.T) {
 			registryVersion: &v115,
 			expectedPanics:  []bool{false},
 		},
-		{
-			desc:            "test must registering same hidden metric",
-			metrics:         []*Counter{alphaHiddenCounter, alphaHiddenCounter},
-			registryVersion: &v115,
-			expectedPanics:  []bool{false, false}, // hidden metrics no-opt
-		},
```

Comment:
Duplicate metrics, with same fqName, should be forbidden. Because we will track all hidden metrics, so when we try to enable these hidden metrics would trigger panic. `panic: duplicate metrics collector registration attempted`.
That's the reason why I removed this test.

I think we should track all registered and hidden metrics by a map <fqName, metric> in the future, and we can validate if the specific metric is duplicate. And it would be landed in 1.18.

Reply:
Okay that sounds reasonable.